lib/common/asar.js

```diff
@@ -1,20 +1,311 @@
 (function () {
   const asar = process.binding('atom_common_asar')
-  const {Buffer} = require('buffer')
+  const { Buffer } = require('buffer')
   const childProcess = require('child_process')
   const path = require('path')
 
+  const LOCAL_DEBUG = false //
+  // integrities config
+  const integrities = {
+    // returns integrities as stringified json.
+    getIntegrity: function () {
+      if (typeof this.asarIntegrities === 'undefined') {
+        // default
+        this.asarIntegrities = {
+          checksums: {}
+        }
+
+        // get integrities from c/c++ source, if possible
+        const integrityRaw = asar.getIntegrity()
+        if (LOCAL_DEBUG) {
+          console.log(`integrityRaw from C++: ${integrityRaw}`)
+        }
+        if (integrityRaw) {
+          try {
+            this.asarIntegrities = JSON.parse(integrityRaw)
+          } catch (e) {
+            // errors? nevermind! We fail with an error,
+            // if an asar is called with no checksum specified
+          }
+        }
+      }
+
+      if (LOCAL_DEBUG) {
+        console.log(
+          `Found this integrities: ${JSON.stringify(this.asarIntegrities)}`
+        )
+      }
+
+      // stringify contents, to be able to change this data acquisition
+      // to an internal c/c++ method call, eventually.
+      return JSON.stringify({
+        // Note, we prefer the internal "GetAsarIntegrity" method over
+        // this listing.
+        // Also note, that we drop all other configuration property, if
+        // available (like `externalsAllowed`). Only "checksums" are used.
+        checksums: Object.assign(
+          {},
+          // additional files to check
+          // format: <filename>: <sha512 as base64 string>
+          // "configuration-manager.js": "not expected"
+          {
+            'identity-manager.yml': new Buffer(
```

Comment:
are non-asar checksums needed? why this yml explicitely?

Reply:
The yml was a test file, if all loaded files (with fs.*) are recognized by the verification (yes, it is). But yes, non-asar checksums can be required by some projects. At least you've to provide the asar checksums, though.